var/spack/repos/builtin/packages/py-pyfr/package.py

```diff
@@ -52,7 +56,49 @@ class PyPyfr(PythonPackage):
 
     # Optional dependencies
     depends_on("py-scipy", when="+scipy", type=("build", "run"))
+    depends_on("metis@5.0:", when="+metis", type=("run"))
     depends_on("scotch@6.0:", when="+scotch", type=("run"))
-    depends_on("cuda@8.0:", when="+cuda", type=("run"))
-    depends_on("rocblas@4.5.0:", when="+hip", type=("run"))
-    depends_on("libxsmm@1.18:+shared blas=0 +large_jit_buffer", when="+libxsmm", type=("run"))
+    depends_on("cuda@8.0: +allow-unsupported-compilers", when="@:1.14.0 +cuda", type=("run"))
+    depends_on("cuda@11.4.0: +allow-unsupported-compilers", when="@1.15.0: +cuda", type=("run"))
+    depends_on("rocblas@5.2.0:", when="+hip", type=("run"))
+    depends_on("libxsmm@1.18:+shared blas=0", when="+libxsmm", type=("run"))
+
+    # Conflicts for compilers not supporting OpenMP 5.1+ from v1.15.0:
+    conflicts("%gcc@:11", when="@1.15.0: +libxsmm", msg="OpenMP 5.1+ supported compiler required!")
+
+    # Explicitly set library path for dependencies
+    def setup_run_environment(self, env):
+        if "+metis" in self.spec:
+            lib_path = join_path(
+                self.spec["metis"].libs.directories[0], "libmetis.{0}".format(dso_suffix)
+            )
```

Comment:
If you want the name of the library and not the directory, that's even easier:
```suggestion
            lib_path = self.spec["metis"].libs[0]
```
Same below

Reply:
I think this is a good solution:
https://github.com/spack/spack/blob/639f7ec42ad672629364990bdc20c36cb301cedb/var/spack/repos/builtin/packages/py-pyfr/package.py#L69-L76